### ISSUE TITLE
Track `recent_attesters` and `recent_proposers` in ActiveState

### ISSUE THREAD

**Issue by djrtwo** [ethereum/beacon_chain#7]
# Issue
The current implementation tracks `balance_deltas` in ActiveState, but the spec instead tracks `recent_attesters` and `recent_proposers` to just process what they are owed at the end of each block. This is simpler and clearer.

# Proposed Implementation
- remove `balance_deltas` from ActiveState
- add `'recent_attesters': ['int24']` to ActiveState
- add `'recent_proposers': [RecentProposerRecord]` to ActiveState
- Add `RecentProposerRecord` object with the following fields:
```python
fields = {
    # Proposer index
    'index': 'int24',
    # New RANDAO commitment
    'randao_commitment': 'hash32',
    # Balance delta
    'balance_delta': 'int24'
}
```
- Process balance deltas at start of epoch based upon recent_attesters and recent_proposers. See spec for more details
- add testing.

**Comment by djrtwo** [comment]
I'm working on this

**Comment by djrtwo** [comment]
addressed in #16
